diff -ruN roscpp-original/src/libros/init.cpp roscpp/src/libros/init.cpp
--- roscpp-original/src/libros/init.cpp	2016-10-27 11:21:44.761163830 +0200
+++ roscpp/src/libros/init.cpp	2016-10-27 11:24:21.418471556 +0200
@@ -64,6 +64,8 @@
 
 #include <cstdlib>
 
+#include <sys/prctl.h>
+
 namespace ros
 {
 
@@ -341,10 +343,15 @@
 
   initInternalTimerManager();
 
+  char name[64];
+  prctl(PR_GET_NAME, name);
+
   TopicManager::instance()->start();
   ServiceManager::instance()->start();
   ConnectionManager::instance()->start();
+  prctl(PR_SET_NAME, "Poll");
   PollManager::instance()->start();
+  prctl(PR_SET_NAME, "XML");
   XMLRPCManager::instance()->start();
 
   if (!(g_init_options & init_options::NoSigintHandler))
@@ -356,6 +363,7 @@
 
   if (!(g_init_options & init_options::NoRosout))
   {
+    prctl(PR_SET_NAME, "LOG");
     g_rosout_appender = new ROSOutAppender;
     ros::console::register_appender(g_rosout_appender);
   }
@@ -411,9 +419,10 @@
   }
 
   if (g_shutting_down) goto end;
-
+  prctl(PR_SET_NAME, "QUEUE");
   g_internal_queue_thread = boost::thread(internalCallbackQueueThreadFunc);
   getGlobalCallbackQueue()->enable();
+  prctl(PR_SET_NAME, name);
 
   ROSCPP_LOG_DEBUG("Started node [%s], pid [%d], bound on [%s], xmlrpc port [%d], tcpros port [%d], using [%s] time", 
 		   this_node::getName().c_str(), getpid(), network::getHost().c_str(), 
